File: src/main/java/Folder.java
import java.util.ArrayList;
import java.util.Arrays;

public class Folder extends AbstractFileSystemNode{
    private ArrayList <AbstractFileSystemNode> children;

    public Folder(String name, AbstractFileSystemNode ... files) {
        this.name = name;
        children = new ArrayList<>();
        children.addAll(Arrays.asList(files));
        for (AbstractFileSystemNode child : children) {
            child.parent = this;
        }
    }

    public ArrayList<File> getFiles() {
        ArrayList<File> files = new ArrayList<>();
        for (AbstractFileSystemNode child : children) {
            if (child instanceof File){
                files.add((File) child);
            }
        }
        return files;
    }
}

File: src/main/java/Folder.kt
import java.util.ArrayList
import java.util.Arrays

class Folder(name: String?, vararg files: AbstractFileSystemNode) : AbstractFileSystemNode() {
    private val children: ArrayList<AbstractFileSystemNode>
    fun getFiles(): ArrayList<File> {
            val files = ArrayList<File>()
            for (child in children) {
                if (child is File) {
                    files.add(child)
                }
            }
            return files
    }

    init {
        this.name_value = name
        children = ArrayList()
        for (child in files) {
            children.add(child)
            child.setParent(this)
        }
    }
}